readme.md
# Logseq Plugin: *Sticky Popup* 📍

- Show selected text in a movable pop-up and positioning calendar.

- Displays user messages by day of the week, with the added functionality of reminding about overdue tasks. When the Journal template is loaded, they are displayed. 2023/08/11🆙

[![latest release version](https://img.shields.io/github/v/release/YU000jp/logseq-plugin-sticky-popup)](https://github.com/YU000jp/logseq-plugin-sticky-popup/releases)
[![License](https://img.shields.io/github/license/YU000jp/logseq-plugin-sticky-popup?color=blue)](https://github.com/YU000jp/logseq-plugin-sticky-popup/LICENSE)
[![Downloads](https://img.shields.io/github/downloads/YU000jp/logseq-plugin-sticky-popup/total.svg)](https://github.com/YU000jp/logseq-plugin-sticky-popup/releases) Published 2023/04/18

## Features

- Pop-ups possible to be moved around and resized.

![demo image](https://user-images.githubusercontent.com/111847207/232673738-4e21395a-b04b-4baf-82cc-c5ff2748bbce.gif)

### Sticky Text (Pop-up)

- Select text and click the same block. Registered in pop-up and automatically locked.

> Markdown is not reflected.

![image](https://user-images.githubusercontent.com/111847207/233500354-a9302519-a0ee-4d0c-b9b9-0b7ffe4bd24e.png)

   > "Unlock" is a button that overwrites the next selected text, and "pin" is a button that saves the position of the popup.

### Sticky Calendar (Pop-up)

<details><summary>Sample</summary>
  
![image](https://user-images.githubusercontent.com/111847207/233500548-4c46d364-5b48-4f23-bf72-1cc6be96e0d2.png)

</details>

- require install [Block Calendar Plugin](https://github.com/vipzhicheng/logseq-plugin-block-calendar) for rendering

- Set `custom`(Widget Location) and `#StickyCalendar`(Provide CSS selector) on the plugin settings

  <details><ummary></summary>
  
  <img src="https://user-images.githubusercontent.com/111847207/232676143-c895a94d-c78e-4c85-8ba1-2cf863813957.png"/>
  
  >Settings of Block Calendar plugin
  
</details>

### Overdue tasks board

- Display overdue tasks from yesterday. (If found)

### Daily message board

- Display user message each day. Supports HTML instead of markdown.

#### As showcase

- Display user favorite image by HTML each day. `<img src="FilePath"/>`
> ⚠️Use direct file paths instead of indirect

## Getting Started

### Install from Logseq Marketplace

- Press [`---`] on the top right toolbar to open [`Plugins`]
- Select `Marketplace`
- Type `popup` in the search field, select it from the search results and install

![image](https://user-images.githubusercontent.com/111847207/232879519-8376669d-3f20-4b28-b0cc-6993c25140b7.png)

### Usage

- Open the plugin settings screen and do the necessary changes. By default, Sticky Calendar is set to "off" in the initial configuration.

### Plugin Settings

#### Sticky Text

- (Sticky Text) Visible or not: select
  - `Journal`
  - `Not-Journal`
  - `All` default
  - `None`
- (Sticky Text) Showing over sidebar or not: toggle
  - `true` default
  - `false`

#### Sticky Calendar

- (Sticky Calendar) Visible or not: select
  - `Journal` default
  - `Not-Journal`
  - `All`
  - `None`
- (Sticky Calendar) Showing over sidebar or not: toggle
  - `true` default
  - `false`

#### Overdue task board

> Enabled when toggles are on. Can be called from the toolbar button.
- Show the board when journal template is loaded: toggle
  `true`
  `false` default
- Show the board when Logseq is loaded: toggle
  `true`
  `false` default

#### Daily message board

> Enabled when toggles are on. Can be called from the toolbar button.
- Show the board when today journal is created: toggle
  `true` default
  `false`
- Show the board when Logseq loaded: toggle
  `true` default
  `false`
- Enable close the board timeout: toggle
  `true` default
  `false`
- Close the board timeout [ms]: select
  - [`8000`, `9000`, `10000` default, `12000`, `14000`, `16000`, `18000`, `20000`]
- Background color (from theme): select
  - `var(--ls-primary-background-color)` default
  - `var(--ls-secondary-background-color)`
  - `var(--ls-tertiary-background-color)`
  - `var(--ls-quaternary-background-color)`
  - `var(--ls-table-tr-even-background-color)`
  - `var(--ls-block-properties-background-color)`
  - `var(--ls-page-properties-background-color)`
- Font Color (from theme): select
  - `var(--ls-primary-text-color)` default
  - `var(--ls-secondary-text-color)`
  - `var(--ls-title-text-color)`
  - `var(--ls-link-text-color)`
- The board title, select localize (your language) or English: select
  - `default` default
  - `en`: English
- Show Monday message board: toggle
  - `true`
  - `false` default
- Message for Monday (Supports HTML instead of markdown)
  - (textarea)
- alias Tuesday...Sunday

## Question

- To reopen popups, press on `📌`button in the toolbar
  - For Daily message board: `💬`
  - For Overdue tasks board: `⏳`

## Showcase / Questions / Ideas / Help

> Go to the [discussion](https://github.com/YU000jp/logseq-plugin-sticky-popup/discussions) tab to ask and find this kind of things.

## Author

- GitHub: [YU000jp](https://github.com/YU000jp)

## Prior art & Credit

### Library

- [@logseq/libs](https://logseq.github.io/plugins/)
- [xyhp915/ moveable plugin UI container](https://github.com/logseq/logseq/pull/3045)

### Icon

- [icooon-mono.com](https://icooon-mono.com/)

---

<a href="https://www.buymeacoffee.com/yu000japan" target="_blank"><img src="https://cdn.buymeacoffee.com/buttons/v2/default-violet.png" alt="🍌Buy Me A Coffee" style="height: 42px;width: 152px" ></a>


CHANGELOG.md
## [1.9.1](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.9.0...v1.9.1) (2023-08-12)


### Bug Fixes

* **doc:** plugin settings ([757e7d9](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/757e7d94ab7870bce8562407a03d927d2a19aa26))
* plugin settings ([8e8b1e2](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/8e8b1e2dd1ce6987a9dd3f17159b3452e60f95b5))

# [1.9.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.8.5...v1.9.0) (2023-08-11)


### Bug Fixes

* **bug:** pinned ([6dddfda](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/6dddfdacbf59cd1914f50fb17ea35428c49eadf6))


### Features

* **Daily Message:** free position and size (Support pinned) ([2dded45](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/2dded45534fda872cd3b663bd10ac3a8266d877f))
* popup button position & when unlock remove recoded text ([e2c24f2](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/e2c24f21fd420bc2855b633bd07ddd158e31734c))
* **Sticky Text:** without markdown ([8d23869](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/8d238691afee52c64fde67ff716526483e050cf2))

## [1.8.5](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.8.4...v1.8.5) (2023-08-09)


### Bug Fixes

* **bug:** not work ([da0e8a1](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/da0e8a187d8b26d441be33cd0b6dbd5f73af96a6))

## [1.8.4](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.8.3...v1.8.4) (2023-06-17)


### Bug Fixes

* **bug:** on click the toolbar button & Cleanup ([b14ae97](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/b14ae97d83a983154ea78e930b135e3bff5295ff))

## [1.8.3](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.8.2...v1.8.3) (2023-05-27)


### Bug Fixes

* **remove:** `Sticky Weekly` ([67ac33e](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/67ac33e8f85fa2d328a60fc65045905e7bef287a))

## [1.8.2](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.8.1...v1.8.2) (2023-05-13)


### Bug Fixes

* **bug:** Popup CSS z-index ([5cc2387](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/5cc238797b8253d72ddcc396b1db696f42257857))

## [1.8.1](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.8.0...v1.8.1) (2023-05-12)


### Bug Fixes

* **bug:** sticky weekly weekday ([c0c5816](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/c0c5816691690bd7d3d832b0a4debc52ef80d6d5))
* **bug:** sticky weekly weekday ([7915890](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/79158900c1943fe2d746c5d5116076246967444a))

# [1.8.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.7.1...v1.8.0) (2023-05-05)


### Features

* **Sticky Weekly:** demo version (for progress only) ([1d58711](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/1d58711187761605e0cb282d725c839dd8acf7fe))

## [1.7.1](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.7.0...v1.7.1) (2023-05-02)


### Bug Fixes

* **bug:** call once onInputSelectionEnd ([f998e73](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/f998e733b6a291678db0460dea6ba188f02e7509))

# [1.7.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.6.1...v1.7.0) (2023-04-25)


### Features

* **feed:** remove graph-related settings and link directly to feed ([f826303](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/f82630371c1f9dda52991a35b5b9325ecbc7e647))

## [1.6.1](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.6.0...v1.6.1) (2023-04-21)


### Bug Fixes

* Fix bug in graph toggle operation (settings item) ([f500847](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/f500847d64775e4d96ae4d2f1951e7ad2d6208ec))

# [1.6.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.5.1...v1.6.0) (2023-04-20)


### Features

* **menu): add tooltip feature fix(page:** handle non-existent page error ([7306b2a](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/7306b2a98a09314dd3ad93766ffe44079b55748b))

## [1.5.1](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.5.0...v1.5.1) (2023-04-19)


### Bug Fixes

* pin function not working ([b3fc553](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/b3fc553cd5fbff6c1fa4f4785e16e43baa647609))

# [1.5.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.4.0...v1.5.0) (2023-04-19)


### Features

* **release:** Fix issues with Unlock page, change to button format, update default settings ([ce16479](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/ce16479674777a67850a26918c269444f15d53c6))
  - Modified ”Unlock” so that it no longer overlaps with long page names
  - Updated the default settings
  - Created an initial display for first-time user
  - Modified the process for refreshing calendar display

# [1.4.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.3.0...v1.4.0) (2023-04-18)


### Features

* Fix bugs on plugin release ([352ed7c](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/352ed7cca838b465e996becfaab10ef6bc6808c3))
* Fix bugs on plugin release ([1ebdee4](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/1ebdee441ddf6e3c6cdc8ab9bbc82f7bc66cfe35))

# [1.4.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.3.0...v1.4.0) (2023-04-18)


### Features

* Fix bugs on plugin release ([1ebdee4](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/1ebdee441ddf6e3c6cdc8ab9bbc82f7bc66cfe35))

# [1.3.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.2.0...v1.3.0) (2023-04-18)


### Features

* **calendar:** enable rendering when toolbar is opened ([9a58c8c](https://github.com/YU000jp/logseq-plugin-sticky-popup/commit/9a58c8c346f9130956264084d320ed11114d8add))

# [1.2.0](https://github.com/YU000jp/logseq-plugin-sticky-popup/compare/v1.1.0...v1.2.0) (2023-04-18)


### Features


src/settings.ts
import { SettingSchemaDesc } from "@logseq/libs/dist/LSPlugin.user";
//user setting
//https://logseq.github.io/plugins/types/SettingSchemaDesc.html
export const settingsTemplate: SettingSchemaDesc[] = [
  {
    key: "headerForStickyText",
    title: "--- Sticky Text ---",
    type: "heading",
    default: "",
    description: "Select string and click the same block. Registered in pop-ups and automatically locked. Markdown is not reflected. Can be called from 📌 button on toolbar.",
  },
  {
    key: "stickyTextVisible",
    title: "(Sticky Text) Visible or not",
    type: "enum",
    enumChoices: ["Journal", "Not-Journal", "All", "None"],
    default: "All",
    description: "",
  },
  {
    key: "stickyTextZIndex",
    title: "(Sticky Text) Show over sidebar or not",
    type: "boolean",
    default: true,
    description: "",
  },
  {
    key: "headerForStickyCalendar",
    title: "--- Sticky Calendar ---",
    type: "heading",
    default: "",
    description: "Require rendering of Block Calendar Plugin. Set `custom` and `#StickyCalendar`(Provide CSS selector) on the plugin settings.",
  },
  {
    key: "stickyCalendarVisible",
    title: "(Sticky Calendar) Visible or not",
    type: "enum",
    enumChoices: ["Journal", "Not-Journal", "All", "None"],
    default: "Journal",
    description: "",
  },
  {
    key: "stickyCalendarZIndex",
    title: "(Sticky Calendar) Show over sidebar or not",
    type: "boolean",
    default: true,
    description: "",
  },
  {
    key: "headerForStickyOverdue",
    title: "--- Overdue task board ---",
    type: "heading",
    default: "",
    description: `
    Display overdue tasks from yesterday (If found).
    Enabled when toggles are on. Can be called from ⏳ button on toolbar.`
    ,
  },
  {
    key: "enableOverdueOnJournalTemplate",
    type: "boolean",
    title: "Show the board when journal template is loaded",
    description: "",
    default: false,
  },
  {
    key: "enableOverdueLogseqLoaded",
    type: "boolean",
    title: "Show the board when Logseq is loaded",
    description: "",
    default: false,
  },
  {
    key: "headerForStickyMessage",
    title: "--- Daily message board ---",
    type: "heading",
    default: "",
    description: `
    Display user message each day.
    Enabled when toggles are on. Can be called from 💬 button on toolbar.

    ⚠️Supports HTML instead of markdown. Use direct file paths instead of indirect.
    `,
  },
  {
    //messageBoard
    key: "enableMessageBox",
    type: "boolean",
    title: "Show the board when today journal is created",
    description: "",
    default: true,
  },
  {
    //messageBoard Logseq loaded
    key: "enableMessageBoxLogseqLoaded",
    type: "boolean",
    title: "Show the board when Logseq loaded",
    description: "",
    default: true,
  },
  {
    key: "enableMessageBoxTimeout",
    type: "boolean",
    title: "Enable close the board timeout",
    description: "",
    default: true,
  },
  {
    //close timeout
    key: "messageBoxTimeout",
    type: "enum",
    title: "Close the board timeout [ms]",
    enumChoices: ["8000", "9000", "10000", "12000", "14000", "16000", "18000", "20000"],
    description: "default: 10000",
    default: "10000",
  },
  {
    //background color
    key: "backgroundColor",
    type: "enum",
    title: "Background color (from theme)",
    enumChoices: [
      "var(--ls-primary-background-color)",
      "var(--ls-secondary-background-color)",
      "var(--ls-tertiary-background-color)",
      "var(--ls-quaternary-background-color)",
      "var(--ls-table-tr-even-background-color)",
      "var(--ls-block-properties-background-color)",
      "var(--ls-page-properties-background-color)",
    ],
    description: "default: var(--ls-primary-background-color)",
    default: "var(--ls-primary-background-color)",
  },

  {
    //color
    key: "fontColor",
    type: "enum",
    title: "Font Color (from theme)",
    enumChoices: [
      "var(--ls-primary-text-color)",
      "var(--ls-secondary-text-color)",
      "var(--ls-title-text-color)",
      "var(--ls-link-text-color)",
    ],
    description: "default: var(--ls-primary-text-color)",
    default: "var(--ls-primary-text-color)",
  },
  {
    key: "localizeOrEnglish",
    type: "enum",
    title: "The board title, select localize (your language) or English",
    enumChoices: ["default", "en"],
    description: "",
    default: "default",
  },
  {
    key: "toggleMonday",
    type: "boolean",
    title: "Show Monday message board",
    description: "",
    default: false,
  },
  {
    key: "monday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Monday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
  {
    key: "toggleTuesday",
    type: "boolean",
    title: "Show Tuesday message board",
    description: "",
    default: false,
  },
  {
    key: "tuesday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Tuesday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
  {
    key: "toggleWednesday",
    type: "boolean",
    title: "Show Wednesday message board",
    description: "",
    default: false,
  },
  {
    key: "wednesday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Wednesday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
  {
    key: "toggleThursday",
    type: "boolean",
    title: "Show Thursday message board",
    description: "",
    default: false,
  },
  {
    key: "thursday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Thursday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
  {
    key: "toggleFriday",
    type: "boolean",
    title: "Show Friday message board",
    description: "",
    default: false,
  },
  {
    key: "friday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Friday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
  {
    key: "toggleSaturday",
    type: "boolean",
    title: "Show Saturday message board",
    description: "",
    default: false,
  },
  {
    key: "saturday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Saturday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
  {
    key: "toggleSunday",
    type: "boolean",
    title: "Show Sunday message board",
    description: "",
    default: false,
  },
  {
    key: "sunday",
    type: "string",
    inputAs: "textarea",
    title: "Message (HTML) for Sunday ⚠️",
    description: "To view the modified content, toggle it off and then on again.",
    default: "",
  },
];

